monero/utils.go

```diff
@@ -22,24 +21,25 @@ var (
 // WaitForBlocks waits for `count` new blocks to arrive.
 // It returns the height of the chain.
 func WaitForBlocks(ctx context.Context, client WalletClient, count int) (uint64, error) {
-	startHeight, err := client.GetChainHeight()
+	c := client.(*walletClient)
```

Comment:
is `*walletClient` the only implementation of `WalletClient`?

Reply:
Yes. I could even see an argument for removing the interface and capitalizing the struct type.

Reply:
yeah I'd be fine with that. or move to interface to the packages where it's used if you want to do any mock testing

Reply:
Let's play with the idea in a different PR. I think it is too big of a change to add into this PR at this point.